core/src/impl/Kokkos_Core.cpp

```diff
@@ -282,11 +295,206 @@ void initialize_profiling(const InitArguments& args) {
     std::exit(EXIT_SUCCESS);
   }
   Kokkos::Tools::parseArgs(args.tool_args);
+  for (const auto& category_values : Kokkos::Impl::metadata_map) {
+    for (const auto& key_value : category_values.second) {
+      Kokkos::Tools::declareMetadata(key_value.first, key_value.second);
+    }
+  }
 }
-
+#define KOKKOS_METADATA_QUOTE_HELPER(x) #x
+#define KOKKOS_METADATA_QUOTE(x) KOKKOS_METADATA_QUOTE_HELPER(x)
```

Comment:
I would prefer defining
```
inline std::string kokkos_git_revision() { return "@KOKKOS_GIT_REVISION@"; }
```
in `cmake/KokkosCore_config.h.in` (or a separate file that we configure with `CMake`) over all these preprocessor defines.

Reply:
Me too. I was copying the other things, but if you're game for this, I'm more than happy to make the change

Reply:
We deferred this feature, based on Damien's comments. When we revisit, I'll likely use this suggestion, thanks